### ISSUE TITLE
string_lit_as_bytes lint isn't trigged when string contains escaped characters

### ISSUE THREAD

**Issue by upsuper** [rust-lang/rust-clippy#4796]
I encountered a case where `string_lit_as_bytes` is not triggered as expected:
```rust
let bs = "a byte string\t\n".as_bytes();
```

If you remove `\t\n`, then it correctly suggests that it can use byte string literal instead, however, with either `\t` or `\n` presents, it doesn't say anything.

I'm testing using Clippy from playground which has version `0.0.212 (2019-10-28 c8e3cfb)`.

**Comment by upsuper** [comment]
Probably not too hard to fix, but not trivial either: https://github.com/rust-lang/rust-clippy/blob/f3a207e63bc9d305c8c908fafb1f34565efc1426/clippy_lints/src/strings.rs#L177-L180

**Comment by hellow554** [comment]
`lit_content.as_str().chars().all(|c| c.is_ascii())` should be `lit_content_as_str().is_ascii()` while we're at it
